superset/views/core.py

```diff
@@ -1788,14 +1789,15 @@ def dashboard(self, dashboard_id):
         dash = qry.one_or_none()
         if not dash:
             abort(404)
-        datasources = set()
+
+        datasources = defaultdict(list)
         for slc in dash.slices:
             datasource = slc.datasource
             if datasource:
-                datasources.add(datasource)
+                datasources[datasource].append(slc)
 
         if config["ENABLE_ACCESS_REQUEST"]:
-            for datasource in datasources:
+            for datasource in datasources.keys():
```

Comment:
Nit. Remove the `.keys()` as this is implicit when iterating over a dictionary.

Reply:
someday i'll learn python